Titanium and plastitanium shards and weapons + missing textures.

Comxy opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Ever been bothered by why titanium glass and plastitanium glass do not drop their own shard types? Well this is the perfect PR for you! Titanium and plastitanium glass shards never existed, and it is probably because the person who made glass way back in the day didn't have time to add these shards. Luckily I decided to add them after all this time. Every piece of code created has been carefully considered and copied form other code, so then you know it is good code. Also I added more tags, I looked at the guidelines and found that adding the fix and qol tags probably boosts my pr score so it will get merged.
![promo1](https://user-images.githubusercontent.com/25363960/165405164-c2de794c-c70a-41aa-b9d0-26ad58c3bfa7.PNG)
![promo2](https://user-images.githubusercontent.com/25363960/165405171-1c9b210c-99e3-4ad5-927a-5cf5338aff4c.PNG)


<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
- Adds titanium glass shards
- Adds plastitanium glass shards
- Adds plasma glass shiv
- Adds titanium glass shiv
- Adds plastitanium glass shiv
- Adds titanium glass spear
- Adds plastitanium glass spear
- Adds time for the creation of shards in stronger glass types

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
fix: When holding plasma glass, the sprite of normal glass was shown in hand since plasma glass in hand didn't exist.
add: Added titanium and plastitanium glass shards as well as spears and shivs for their types, including plasma glass shivs.
imageadd: Sprites for plasma, titanium and plastitanium glass shards when hold in hand.
qol: titanium and plastitanium glass having their own shards makes sense, and is a quality feature.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Comxy committed:
Adds titanium and plastitanium shards and weapons, and well as missing plasma glass textures.

TemporalOroboros commented:
As someone who's static is digi I am in pain, but I approve nonetheless.

tralezab commented on code/game/objects/items/stacks/sheets/glass.dm:
```diff
@@ -377,6 +377,64 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
 	weld_material = /obj/item/stack/sheet/plasmaglass
+
+/obj/item/shard/plasma/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/stack/sheet/cloth))
+		var/obj/item/stack/sheet/cloth/C = I
+		to_chat(user, span_notice("You begin to wrap the [C] around the [src]..."))
+		if(do_after(user, 70, target = src))
+			var/obj/item/knife/shiv/plasma/S = new /obj/item/knife/shiv/plasma
+			C.use(1)
+			to_chat(user, span_notice("You wrap the [C] around the [src] forming a makeshift weapon."))
+			remove_item_from_storage(src)
+			qdel(src)
+			user.put_in_hands(S)
```
you copy paste this entire thing three times, just make it part of `/obj/item/shard`. Keep it DRY! Don't Repeat Yourself!

tralezab commented on code/game/objects/items/stacks/sheets/glass.dm:
```diff
@@ -377,6 +377,64 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
 	icon_prefix = "plasma"
 	weld_material = /obj/item/stack/sheet/plasmaglass
+
+/obj/item/shard/plasma/attackby(obj/item/I, mob/user, params)
```
a better variable name than `I`, please

tralezab commented on code/game/objects/items/stacks/sheets/glass.dm:
```diff
@@ -377,6 +377,64 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
+
+/obj/item/shard/plasma/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/stack/sheet/cloth))
+		var/obj/item/stack/sheet/cloth/C = I
```
a better variable name than `C`, please

tralezab commented on code/game/objects/items/stacks/sheets/glass.dm:
```diff
@@ -377,6 +377,64 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
+		var/obj/item/stack/sheet/cloth/C = I
+		to_chat(user, span_notice("You begin to wrap the [C] around the [src]..."))
+		if(do_after(user, 70, target = src))
+			var/obj/item/knife/shiv/plasma/S = new /obj/item/knife/shiv/plasma
```
a better variable name than `S`, please

tralezab commented on code/game/objects/items/stacks/sheets/glass.dm:
```diff
@@ -377,6 +377,64 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
+	icon_prefix = "titanium"
+	weld_material = /obj/item/stack/sheet/titaniumglass
+
+/obj/item/shard/titanium/attackby(obj/item/I, mob/user, params)
```
these have the same issues but you'll just move this into `/obj/item/shard` behavior anyway

tralezab commented on code/game/objects/items/stacks/sheets/glass.dm:
```diff
@@ -377,6 +377,64 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
+	icon_prefix = "plastitanium"
+	weld_material = /obj/item/stack/sheet/plastitaniumglass
+
+/obj/item/shard/plastitanium/attackby(obj/item/I, mob/user, params)
```
these have the same issues but you'll just move this into `/obj/item/shard` behavior anyway

tralezab reviewed (changes requested):
Codewise, some things need to be cleaned up

tralezab commented:
On the balance side of things I worry a little bit about reintroducing "spears breaking doors" meta with the plasmashard spear breaking doors when thrown but maybe it's annoying enough to get that it's fine. I dunno if anyone here has tried to break a plasmaglass window but it fuckin blooooows

itseasytosee commented:
Ungodly

TheBonded commented:
are titanium spears having 1 higher throwforce than plasma spears (but 1 lower force) intended?

thgvr commented:
tralezab, we already have spears that can break airlocks. Bamboo spears have 22 throwforce, same as these new ones

Comxy committed:
Small balance & stats update

Comxy commented:
I changed some things. I made sure titanium glass spears cannot break doors, and increased some other stats to make it different. Also, you cannot get plastitanium shards anymore by crafting with sheets, since it is too hard to break. Now you can get is by breaking a plastitanium window instead, so it is harder to get. Also I fixed the code up.

bruhlookatthisdood commented:
i have never seen a "large shard"

VladinXXV commented on code/game/objects/items/knives.dm:
```diff
@@ -156,6 +156,49 @@
+	lefthand_file = 'icons/mob/inhands/weapons/swords_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/swords_righthand.dmi'
+	desc = "A makeshift titanium infused glass shiv."
+	force = 8
```
Isn't it redundant to have `force = 8` here? It inherits the exact same damage from the base shiv, no? Same stuff for the following in all three of them:
`icon = 'icons/obj/shards.dmi'`
`lefthand_file = 'icons/mob/inhands/weapons/swords_lefthand.dmi'`
`righthand_file = 'icons/mob/inhands/weapons/swords_righthand.dmi'`

VladinXXV reviewed (commented):
this pr is swag

TheBonded commented:
> i have never seen a "large shard"

they dont have any actual differences other than their sprite as far as i know, youve probably seen one and just didnt pay attention since its still just called "Glass Shard" in game

Comxy committed:
Overlooked inheriting change

Comxy commented:
It should be ready to get merged, the code is fine now, the balancing is very good, feel free to merge it any day.

TheBonded commented:
> It should be ready to get merged, the code is fine now, the balancing is very good, feel free to merge it any day.

![image](https://user-images.githubusercontent.com/58570888/165836584-7e590aa7-726d-4f64-87af-e2451ae53ca0.png)
## when your pr is ready to be re-reviewed and merged:

in the top of the pr, where it shows reviewers, be sure to press the little circle to re-request a review from anyone who requested changes when you are ready for the pr to be checked and merged again.

## that way you dont end up with ugly red circles on your PR like this!
![image](https://user-images.githubusercontent.com/58570888/165836836-cad3c8cc-f83b-4cf9-87ea-0b618321441a.png)

TheBonded commented:
(this is obviously not required for your pr to be merged but it does look nice and notify the people who reviewed that you requested them to come back and review the pr again)

TheBonded commented:
resolves https://github.com/tgstation/tgstation/issues/66658

Comxy committed:
plastitanium shiv force fix

bruhlookatthisdood commented:
(why is the plasma shards slightly darker than the plasma shiv)

Comxy commented:
> (why is the plasma shards slightly darker than the plasma shiv)

That is probably shading and there are more cracks in the shiv one.

Comxy committed:
saturation fix

Comxy committed:
adds plasma shard to trash

ATH1909 commented on code/game/objects/items/knives.dm:
```diff
@@ -156,6 +156,41 @@
+	name = "titanium shiv"
+	icon_state = "titaniumshiv"
+	inhand_icon_state = "titaniumshiv"
+	desc = "A makeshift titanium infused glass shiv."
```
```suggestion
	desc = "A makeshift titanium-infused glass shiv."
```

ATH1909 commented on code/game/objects/items/knives.dm:
```diff
@@ -156,6 +156,41 @@
+	name = "plasma shiv"
+	icon_state = "plasmashiv"
+	inhand_icon_state = "plasmashiv"
+	desc = "A makeshift plasma glass shiv."
```
```suggestion
	desc = "A makeshift plasmaglass shiv."
```

ATH1909 commented on code/game/objects/items/knives.dm:
```diff
@@ -156,6 +156,41 @@
+	name = "plastitanium shiv"
+	icon_state = "plastitaniumshiv"
+	inhand_icon_state = "plastitaniumshiv"
+	desc = "A makeshift titanium infused plasma glass shiv."
```
```suggestion
	desc = "A makeshift titanium-infused plasmaglass shiv."
```

ATH1909 commented on code/game/objects/items/stacks/sheets/glass.dm:
```diff
@@ -77,7 +77,7 @@ GLOBAL_LIST_INIT(glass_recipes, list ( \
 	new/datum/stack_recipe("directional window", /obj/structure/window/plasma/unanchored, time = 0, on_floor = TRUE, window_checks = TRUE), \
 	new/datum/stack_recipe("fulltile window", /obj/structure/window/plasma/fulltile/unanchored, 2, time = 0, on_floor = TRUE, window_checks = TRUE), \
-	new/datum/stack_recipe("plasma glass shard", /obj/item/shard/plasma, time = 0, on_floor = TRUE) \
+	new/datum/stack_recipe("plasma glass shard", /obj/item/shard/plasma, time = 20, on_floor = TRUE) \
```
```suggestion
	new/datum/stack_recipe("plasmaglass shard", /obj/item/shard/plasma, time = 20, on_floor = TRUE) \
```

ATH1909 commented on code/game/objects/items/stacks/sheets/glass.dm:
```diff
@@ -332,20 +331,45 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
+			if(do_after(user, 140, target = src))
+				var/obj/item/knife/shiv/plasma/shiv = new /obj/item/knife/shiv/plasma
+				cloth.use(1)
+				to_chat(user, span_notice("You wrap the [cloth] around the [src] forming a makeshift weapon."))
```
```suggestion
				to_chat(user, span_notice("You wrap the [cloth] around the [src], forming a makeshift weapon."))
```

ATH1909 commented on code/game/objects/items/stacks/sheets/glass.dm:
```diff
@@ -332,20 +331,45 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
+			if(do_after(user, 70, target = src))
+				var/obj/item/knife/shiv/titanium/shiv = new /obj/item/knife/shiv/titanium
+				cloth.use(1)
+				to_chat(user, span_notice("You wrap the [cloth] around the [src] forming a makeshift weapon."))
```
```suggestion
				to_chat(user, span_notice("You wrap the [cloth] around the [src], forming a makeshift weapon."))
```

ATH1909 commented on code/game/objects/items/stacks/sheets/glass.dm:
```diff
@@ -332,20 +331,45 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
+			if(do_after(user, 70, target = src))
+				var/obj/item/knife/shiv/plastitanium/shiv = new /obj/item/knife/shiv/plastitanium
+				cloth.use(1)
+				to_chat(user, span_notice("You wrap the [cloth] around the [src] forming a makeshift weapon."))
```
```suggestion
				to_chat(user, span_notice("You wrap the [cloth] around the [src], forming a makeshift weapon."))
```

ATH1909 commented on code/game/objects/items/stacks/sheets/glass.dm:
```diff
@@ -332,20 +331,45 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
+			if(do_after(user, 35, target = src))
+				var/obj/item/knife/shiv/shiv = new /obj/item/knife/shiv
+				cloth.use(1)
+				to_chat(user, span_notice("You wrap the [cloth] around the [src] forming a makeshift weapon."))
```
```suggestion
				to_chat(user, span_notice("You wrap the [cloth] around the [src], forming a makeshift weapon."))
```

Comxy committed:
merge conflict and small changes fix

Comxy committed:
Trying to resolve merge conflict pain

Ebb-Real commented:
This could probably be turned into GAGS if you're up for it. Would maybe be a bit difficult to get the texture and color right though.

Comxy commented:
> This could probably be turned into GAGS if you're up for it. Would maybe be a bit difficult to get the texture and color right though.

What do you mean with GAGS?

Ebb-Real commented:
> > This could probably be turned into GAGS if you're up for it. Would maybe be a bit difficult to get the texture and color right though.
> 
> What do you mean with GAGS?

GAGS is a way to assemble a sprite with layers and assigned colors. This means if you have multiple sprites that are just recolors of the original you can turn them into GAGS and just have one or two sprites that is white with shading and assign colors to them with code and it will color it for you, the assigned color's brightness will be adjusted by the shading you added to the GAGS sprite.

This is good because its easy to recolor sprites (in this case if someone adds a new type of glass they just gotta assign the color of the new glass) and because you need less sprites that are just variations of the first one, so you can just have one sprite that covers them all. GAGS also allows for sprites to be recolored mid-game this is how the color customizable items from vendors work, these include: bandana, beret and shoes.

Here's an example: For bandanas you would need a separate sprite for all of these
![image](https://user-images.githubusercontent.com/95765134/167294628-c4a3519e-31e4-4ca7-8681-1729316a9267.png)

but instead you just need these sprites and can assemble the layers in different ways and different colors:
![image](https://user-images.githubusercontent.com/95765134/167294703-8f1aec79-84f4-4c5d-a9bb-428f475925f2.png)
![image](https://user-images.githubusercontent.com/95765134/167294720-bb463897-b4da-4587-8015-c761f5af0f3a.png)
![image](https://user-images.githubusercontent.com/95765134/167294729-d8f0af03-85ed-4a54-9269-2e191d2634f2.png)



Here's a guide on how to do it: https://hackmd.io/@tgstation/GAGS-Walkthrough

Comxy commented:
GAGS seem like a cool idea, but I will pass on it for now. Maybe someone else will do it or I'll revisit this idea but I'll see if this PR gets merged first as it is right now.

tralezab commented on code/game/objects/items/stacks/sheets/glass.dm:
```diff
@@ -332,20 +331,45 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
+	else if(istype(item, /obj/item/stack/sheet/cloth))
+		var/obj/item/stack/sheet/cloth/cloth = item
+		to_chat(user, span_notice("You begin to wrap the [cloth] around the [src]..."))
+		if(istype(src, /obj/item/shard/plasma))
+			if(do_after(user, 140, target = src))
+				var/obj/item/knife/shiv/plasma/shiv = new /obj/item/knife/shiv/plasma
+				cloth.use(1)
+				to_chat(user, span_notice("You wrap the [cloth] around the [src], forming a makeshift weapon."))
+				remove_item_from_storage(src)
+				qdel(src)
+				user.put_in_hands(shiv)
+		else if(istype(src, /obj/item/shard/titanium))
+			if(do_after(user, 70, target = src))
+				var/obj/item/knife/shiv/titanium/shiv = new /obj/item/knife/shiv/titanium
+				cloth.use(1)
+				to_chat(user, span_notice("You wrap the [cloth] around the [src], forming a makeshift weapon."))
+				remove_item_from_storage(src)
+				qdel(src)
+				user.put_in_hands(shiv)
+		else if(istype(src, /obj/item/shard/plastitanium))
+			if(do_after(user, 70, target = src))
+				var/obj/item/knife/shiv/plastitanium/shiv = new /obj/item/knife/shiv/plastitanium
+				cloth.use(1)
+				to_chat(user, span_notice("You wrap the [cloth] around the [src], forming a makeshift weapon."))
+				remove_item_from_storage(src)
+				qdel(src)
+				user.put_in_hands(shiv)
+		else
+			if(do_after(user, 35, target = src))
+				var/obj/item/knife/shiv/shiv = new /obj/item/knife/shiv
+				cloth.use(1)
+				to_chat(user, span_notice("You wrap the [cloth] around the [src], forming a makeshift weapon."))
+				remove_item_from_storage(src)
+				qdel(src)
+				user.put_in_hands(shiv)
```
Just have the shard have a var for what it should turn into when you apply cloth, and then we don't need to have all this repeating code. do_after times can also be like that

tralezab commented on code/game/objects/items/stacks/sheets/glass.dm:
```diff
@@ -332,20 +331,45 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
+		var/obj/item/stack/sheet/cloth/cloth = item
+		to_chat(user, span_notice("You begin to wrap the [cloth] around the [src]..."))
+		if(istype(src, /obj/item/shard/plasma))
+			if(do_after(user, 140, target = src))
```
```suggestion
			if(do_after(user, 14 SECONDS, target = src))
```

tralezab commented on code/game/objects/items/stacks/sheets/glass.dm:
```diff
@@ -332,20 +331,45 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
+				qdel(src)
+				user.put_in_hands(shiv)
+		else if(istype(src, /obj/item/shard/titanium))
+			if(do_after(user, 70, target = src))
```
```suggestion
			if(do_after(user, 7 SECONDS, target = src))
```

tralezab commented on code/game/objects/items/stacks/sheets/glass.dm:
```diff
@@ -332,20 +331,45 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
+				qdel(src)
+				user.put_in_hands(shiv)
+		else if(istype(src, /obj/item/shard/plastitanium))
+			if(do_after(user, 70, target = src))
```
```suggestion
			if(do_after(user, 7 SECONDS, target = src))
```

tralezab commented on code/game/objects/items/stacks/sheets/glass.dm:
```diff
@@ -332,20 +331,45 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
+				qdel(src)
+				user.put_in_hands(shiv)
+		else
+			if(do_after(user, 35, target = src))
```
```suggestion
			if(do_after(user, 3.5 SECONDS, target = src))
```

tralezab reviewed (changes requested):
okay lets go on to actual code stuffs

Comxy committed:
Getting 900IQ code skills

Comxy committed:
always places craft_time under shiv_type

tralezab reviewed (approved):
awesome possum
